nemo/collections/asr/parts/submodules/conformer_modules.py

```diff
@@ -351,13 +370,25 @@ def __init__(
             raise ValueError(f"conv_norm_type={norm_type} is not valid!")
 
         self.activation = Swish()
-        self.pointwise_conv2 = nn.Conv1d(
-            in_channels=dw_conv_input_dim, out_channels=d_model, kernel_size=1, stride=1, padding=0, bias=True
-        )
+        if self.conv_pointwise_type == 'conv1d':
+            self.pointwise_conv2 = nn.Conv1d(
+                in_channels=dw_conv_input_dim, out_channels=d_model, kernel_size=1, stride=1, padding=0, bias=True
+            )
+        elif self.conv_pointwise_type == 'linear':
```

Comment:
I wonder. Conv1d with kernel 1 is identical to linear layer, so what if we do something like silently convert the conv kernel shape to linear compatible weights and don't use explicitly the Lineae layer but use functional.linear() instead ? 

My assumption is we don't want to break checkpoint compatibility - and we don't need to. We can take a model trained on conv1d and use the weight in a F.linear() after weight conversion - this way no checkpoint incompatibility is there for old models but still there is speedup